beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2201,9 +2203,10 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
     ) -> Result<GossipVerifiedDataColumn<T>, GossipDataColumnError> {
         metrics::inc_counter(&metrics::DATA_COLUMN_SIDECAR_PROCESSING_REQUESTS);
         let _timer = metrics::start_timer(&metrics::DATA_COLUMN_SIDECAR_GOSSIP_VERIFICATION_TIMES);
-        GossipVerifiedDataColumn::new(data_column_sidecar, subnet_id, self).inspect(|_| {
-            metrics::inc_counter(&metrics::DATA_COLUMN_SIDECAR_PROCESSING_SUCCESSES);
-        })
+        GossipVerifiedDataColumn::new(data_column_sidecar, subnet_id, CheckBlockHeader::No, self)
```

Comment:
```suggestion
        GossipVerifiedDataColumn::new(data_column_sidecar, subnet_id, CheckBlockHeader::Yes, self)
```
We need to check header for gossip columns. We can only skip the header check if we're constructing a `DataColumnSidecar` from an already verified `SignedBeaconBlockHeader`.